core/zms/src/main/rdl/Role.rdli

```diff
@@ -129,6 +129,19 @@ resource DomainRoleMembers GET "/domain/{domainName}/member" {
     }
 }
 
+// Fetch all the roles across domains by either calling or specified principal
+resource DomainRoleMember GET "/all_roles?principal={principal}" (name=GetAllRoles) {
```

Comment:
can we also extend this to include an optional domainName attribute where I would only return the role names from the given domain name if specified. We don't have an API that just returns the role names that a given principal is member  in a domain so we don't want them to get the full list and then parse and filter out the domain they're interested in.

Reply:
Sure thing.

Reply:
Done